pennylane/qnode.py

```diff
@@ -1077,7 +1073,7 @@ def _pd_analytic(self, params, idx, force_order2=False, **kwargs):
 
                 # transform the observables
                 obs = [
-                    self._transform_observable(ob, ob_successors, w, Z)
+                    self._transform_observable(ob, w, Z)
```

Comment:
For the old QNode I just implemented the simpler (but probably less efficient) solution, i.e. transforming all the observables. Z transforms all non-descendant observables to zero observables.

Reply:
This is assuming that the old QNode will go away at some point, of course.